### PR TITLE
Core: Discourage calls to multiworld random

### PR DESCRIPTION
**Pull request by alwaysintreble**
## What is this fixing or adding?
Marks calls to `MultiWorld.random` as "deprecated" with a property getter. Uses private attr for the actual object. Removes references to the public random from core and LTTP. Not happy with how I did it with core, but not sure there's a better way to do it, and not really planning to remove the private attr. Next step is to just remove the property.

## How was this tested?
Unit tests. Needs a bunch of practical testing with LTTP as I had to touch a ton of code.

### CODE REVIEW

**Comment by beauxq** [comment]
I think this will make type checkers report an error on every usage or `world._random` that isn't in `MultiWorld`

We could put a `# type: ignore` on all of those...

Or another possibility is to make another property getter with a name that will stick out a little, like `random_core_only`

**Comment by alwaysintreble** [comment]
I want to eventually move the usages in Main out of Main (probably into MultiWorld methods), but that's out of scope for this PR. `random_core_only` feels a bit dirty. PEP 484 isn't perfect; PyCharm seems to ignore it at least, but it's acceptable at least imo.

**Comment by black-sliver** [comment]
The amount of type: ignore is horrible :-D Especially in fille.
I was suggestion earlier that the multiworld could create a new seeded random for outside uses - similar to how it creates a random per player. This would work for Main, but not sure how to handle fill, which is the biggest offender currently.

**Comment by Berserker66** [comment]
Is there a pertinent reason to "Core: Discourage calls to multiworld random" ?

**Comment by alwaysintreble** [comment]
> Is there a pertinent reason to "Core: Discourage calls to multiworld random" ?

As noted here and in other PRs, world's now have their own random object to use at their discretion. Calls to the multiworld random are already not great considering you had to literally ban it from being used during threading, and stage calls are done using a set.

File: worlds/alttp/__init__.py [hunk 1]
```diff
@@ -339,8 +339,8 @@ def create_regions(self):
             world.fix_fake_world[player] = False
 
         # seeded entrance shuffle
-        old_random = world.random
-        world.random = random.Random(self.er_seed)
+        old_random = self.random
+        self.random = random.Random(self.er_seed)
```
[the remark this answers is not in the document]
Comment (alwaysintreble) [reply worlds/alttp/__init__.py:342]: i missed updating most of EntranceShuffle.py apparently. Generated a bunch of seeds and compared the outputs and also generated with the same seeds multiple times to ensure they were still deterministic.

File: worlds/alttp/__init__.py [hunk 2]
```diff
@@ -264,7 +263,7 @@ def stage_assert_generate(cls, multiworld: MultiWorld):
         if not os.path.exists(rom_file):
             raise FileNotFoundError(rom_file)
         if multiworld.is_race:
-            import xxtea
+            pass
```
[the remark this answers is not in the document]
Comment (alwaysintreble) [reply worlds/alttp/__init__.py]: had a setting on where pycharm was auto optimizing imports on commit, but it was optimizing the xxtea import completely out of existence...

**Pull request closed without merging**